RubeesEat/Model/DB/DbCommandExtensions.cs

```diff
@@ -0,0 +1,43 @@
+using System.Data.Common;
+
+namespace RubeesEat.Model.DB;
+
+public static class DbCommandExtensions
+{
+
+    public static DbParameter AddParameter(this DbCommand command, string paramName, Guid paramValue)
+    {
+        var idParam = command.CreateParameter();
+        idParam.ParameterName = paramName;
+        idParam.Value = paramValue;
+        command.Parameters.Add(idParam);
+        return idParam;
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string paramName, decimal paramValue)
+    {
+        var amountParam = command.CreateParameter();
+        amountParam.ParameterName = paramName;
+        amountParam.Value = paramValue;
+        command.Parameters.Add(amountParam);
+        return amountParam;
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string paramName, DateTime paramValue)
+    {
+        var dateParam = command.CreateParameter();
+        dateParam.ParameterName = paramName;
+        dateParam.Value = paramValue;
+        command.Parameters.Add(dateParam);
+        return dateParam;
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string paramName, string paramValue)
+    {
+        var nameParam = command.CreateParameter();
+        nameParam.ParameterName = paramName;
+        nameParam.Value = paramValue;
+        command.Parameters.Add(nameParam);
+        return nameParam;
+    }
```

Comment:
All of these methods do the same thing with the difference being the type of the value. `parameter.Value` is typed as `object` so we could do just one method with an `object value` and remove the remaining three methods.

By specifying the types here as separate values, we have the benefit of allowing only types that we know the parameters work for. So it is not a bad idea to do this to guard against supplying types of values that are not supported at all. If we want to keep the methods for this reason though, we should extract all the duplication into its own private method and call it from the four methods.